File: term-reduce.js
#!/usr/bin/nodejs

var readline = require('readline');

var rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout,
	terminal: false
});

var prevnum = 0, prevterm, prevfile, prevlength;
rl.on('line', function (line) {
	var values = line.split('\t'), term = values[0], file = values[1], num = parseInt(values[2]), length = parseInt(values[3]);
	if (term == prevterm && file == prevfile) {
		prevnum += num;
	} else {
		outputLine(prevterm, prevfile, prevnum, prevlength);
		prevterm = term;
		prevfile = file;
		prevnum = num;
		prevlength = length;
	}
}).on('close', function () {
	outputLine(prevterm, prevfile, prevnum, prevlength);
});

function outputLine(term, file, num, length) {
	num > 0 && console.log([term, file, num, length].join('\t'));
}

File: term-reduce.py
import fileinput

prevterm, prevfile, prevnum, prevlength = '', '', 0, 0

def outputLine(term, file, num, length):
  if (num > 0):
    print("%s\t%s\t%d\t%s\t" % (term, file, num, length))

for line in fileinput.input():
  term, file, num, length = line.split()
  num = int(num)
  if (term == prevterm and file == prevfile):
    prevnum += num
  else:
    outputLine(prevterm, prevfile, prevnum, prevlength)
    prevterm, prevfile, prevnum, prevlength = term, file, num, length

outputLine(prevterm, prevfile, prevnum, prevlength)